File: src/main/java/ch/ibw/appl/apiedi/bimapclient/js/projekt.js
export class Projekt {
    constructor(id, projektnummer, projektname, ort, bauart, projektverfasser, leistungen, koordX, koordY, realisierungsjahr, bausumme, bauherr) {
        this.id = id;
        this.projektnummer = projektnummer;
        this.projektname = projektname;
        this.ort = ort;
        this.bauart = bauart;
        this.projektverfasser = projektverfasser;
        this.leistungen = leistungen;
        this.koordx = koordX;
        this.koordy = koordY;
        this.realisierungsjahr = realisierungsjahr;
        this.bausumme = bausumme;
        this.bauherr = bauherr;
    }
}

Projekt.fromJson = function (json){
    var obj = JSON.parse (json);
    return new Book (obj.idprojekt, obj.projektnummer, obj.projektname, obj.ort, obj.bauart,
        obj.projektverfasser, obj. leistungen, obj.koordx, ob.koordy, obj.realisierungsjahr, obj.bausumme, obj.bauherr);
};

File: src/main/java/ch/ibw/appl/bimap/server/projekte/model/Projekt.java
package ch.ibw.appl.bimap.server.projekte.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class Projekt {

    public int idprojekt;
    public int projektnummer;
    public String projektname;
    public int ortid;
    public int plz;
    public String ort;
    public int koordx;
    public int koordy;
    public int realisierungsjahr;
    public float bausumme;
    public int bauherrid;
    public String bauherr;
    public String bauart;
    public String projektverfasser;
    public String leistungen;

    public static Projekt create(int idprojekt, int projektnummer, String projektname, int ortid, int koordx, int koordy, int realisierungsjahr, float bausumme, int bauherrid) {
        Projekt pr = new Projekt();
        pr.idprojekt = idprojekt;
        pr.projektname = projektname;
        pr.projektnummer = projektnummer;
        pr.ortid = ortid;
        pr.koordx = koordx;
        pr.koordy = koordy;
        pr.realisierungsjahr = realisierungsjahr;
        pr.bausumme = bausumme;
        pr.bauherrid = bauherrid;
        return pr;
    }

    @Override
    public String toString() {
        return "Projekt{" +
                "idprojekt=" + idprojekt +
                ", projektnummer=" + projektnummer +
                ", projektname='" + projektname + '\'' +
                ", plz=" + plz +
                ", ort='" + ort + '\'' +
                ", koordx=" + koordx +
                ", koordy=" + koordy +
                ", realisierungsjahr=" + realisierungsjahr +
                ", bausumme=" + bausumme +
                ", bauherr='" + bauherr + '\'' +
                ", bauart='" + bauart + '\'' +
                ", projektverfasser='" + projektverfasser + '\'' +
                ", leistungen='" + leistungen + '\'' +
                '}';
    }
}
